### PR TITLE
Checkoutanalytics mvp with 3DS2 events

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
- Adding analytics events for the 3DS2 process (data sent, iframe loaded, fingerprint/challenge complete)
- Moved app specific analytics logic to a separate processing function (`analyticsPreProcessor`) in order to separate concerns and make testing easier
- Added `"configured"` type to info event - generate when a component's securedFields are all configured and ready for use
- Added "focus" & "unfocus" info events for the Custom Card component

## Tested scenarios
All existing unit tests run
Added unit tests for all the analytics events that we currently generate

<hr>

**NOTE:** extends changes made in #2521

### CODE REVIEW

File: packages/lib/src/components/UIElement.tsx [hunk 1]
```diff
@@ -78,8 +78,7 @@ export class UIElement<P extends UIElementProps = any> extends BaseElement<P> im
 
         const { type } = analyticsObj;
 
-        // let storedCardIndicator;
-        if (type === ANALYTICS_RENDERED_STR) {
+        if (type === ANALYTICS_RENDERED_STR || type === ANALYTICS_CONFIGURED_STR) {
             // Check if it's a storedCard
```
Comment (longyulongyu) [inline packages/lib/src/components/UIElement.tsx:82]: Would it be possible to let the corresponding children components (e.g. `scheme` and `bcmc`) send the correct data `component` and `analyticsObj`?

I think the UIElement doesn't need to know anything about the children component, whether it's a stored card or not.

File: packages/lib/src/components/ThreeDS2/components/Challenge/types.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export interface DoChallenge3DS2Props extends ChallengeData {
     onCompleteChallenge: (resolveObject: ThreeDS2FlowObject) => void;
     onErrorChallenge: (rejectObject: ThreeDS2FlowObject) => void;
     onActionHandled: (rtnObj: ActionHandledReturnObject) => void;
+    onFormSubmit: (w) => void;
```
Comment (longyulongyu) [inline packages/lib/src/components/ThreeDS2/components/Challenge/types.ts:10]: Could we make the type and name of `w` more specific, i suppose it's an analytic message and the type is `string`?

I see it's being defined in many places, maybe we can create a share `type` for it.

Comment (sponglord) [reply packages/lib/src/components/ThreeDS2/components/Challenge/types.ts:10]: Done

File: packages/lib/src/components/UIElement.tsx (hunk 1 above)
Comment (sponglord) [reply packages/lib/src/components/UIElement.tsx:82]: I agree this is Card specific logic... I'll make the change

**Review by longyulongyu — APPROVED** [review]

**Pull request merged**